agreement/gossip/network.go

```diff
@@ -87,6 +91,13 @@ func (i *networkImpl) processVoteMessage(raw network.IncomingMessage) network.Ou
 }
 
 func (i *networkImpl) processProposalMessage(raw network.IncomingMessage) network.OutgoingMessage {
+	if Trace != nil {
+		hasher := fnv.New64a()
+		hasher.Write(raw.Data)
+		var msgBytes []byte = []byte("prop        ")
+		hash := hasher.Sum(msgBytes[0:4])
+		Trace.Trace(hash)
```

Comment:
Could you move this implementation into the `graphtraceMessageTracer.Trace` ? It would be much cleaner that way.